From: Selvin Xavier <selvin.xavier@broadcom.com>
Date: Thu, 17 Aug 2017 07:58:07 -0700
Subject: RDMA: Fix return value check for ib_get_eth_speed()
Patch-mainline: v4.14-rc1
Git-commit: 7be05753ccc27ce056d45f06a50d150927a88ed7
References: bsc#1050244 FATE#322915

ib_get_eth_speed() return 0 on success. Fixing the condition checking
and prevent reporting failure for query_port verb.

Fixes: d41861942fc5 ("Add generic function to extract IB speed from netdev")
Signed-off-by: Selvin Xavier <selvin.xavier@broadcom.com>
Reviewed-by: Leon Romanovsky <leonro@mellanox.com>
Reviewed-by: Yuval Shaia <yuval.shaia@oracle.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/bnxt_re/ib_verbs.c     |    4 ++--
 drivers/infiniband/hw/usnic/usnic_ib_verbs.c |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/infiniband/hw/bnxt_re/ib_verbs.c
+++ b/drivers/infiniband/hw/bnxt_re/ib_verbs.c
@@ -264,8 +264,8 @@ int bnxt_re_query_port(struct ib_device
 	 * IB stack to avoid race in the NETDEV_UNREG path
 	 */
 	if (test_bit(BNXT_RE_FLAG_IBDEV_REGISTERED, &rdev->flags))
-		if (!ib_get_eth_speed(ibdev, port_num, &port_attr->active_speed,
-				      &port_attr->active_width))
+		if (ib_get_eth_speed(ibdev, port_num, &port_attr->active_speed,
+				     &port_attr->active_width))
 			return -EINVAL;
 	return 0;
 }
--- a/drivers/infiniband/hw/usnic/usnic_ib_verbs.c
+++ b/drivers/infiniband/hw/usnic/usnic_ib_verbs.c
@@ -309,8 +309,8 @@ int usnic_ib_query_port(struct ib_device
 	usnic_dbg("\n");
 
 	mutex_lock(&us_ibdev->usdev_lock);
-	if (!ib_get_eth_speed(ibdev, port, &props->active_speed,
-			      &props->active_width)) {
+	if (ib_get_eth_speed(ibdev, port, &props->active_speed,
+			     &props->active_width)) {
 		mutex_unlock(&us_ibdev->usdev_lock);
 		return -EINVAL;
 	}
